src/connectors/newgrounds.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const artistSelector = '.ng-apg-media-artist';
+const trackSelector = '.ng-apg-media-title';
+
+function setupConnector() {
+	if (isEmbeddedPlayer()) {
+		setupEmbeddedPlayer();
+
+		Connector.getArtistTrack = () => {
+			let { artist, track } = Util.splitArtistTrack(document.getElementsByTagName('h2')[0].innerText, '-');
+
+			if (!artist) {
+				artist = document.getElementsByTagName('h4')[0].innerText;
+				track = document.getElementsByTagName('h2')[0].innerText;
```

Comment:
1. These selectors are ambiguous. I think they should be updated to make sure every selector is unique, e.g. `.item-user h4` for artist (username).
2. `Util.getTextFromSelectors` is prefferable to get text from elements, as it safer to use.